### DISCUSSION TITLE
Difference between actix?

### DISCUSSION THREAD

**Discussion by pickfire** [uazu/stakker#12] [Q&A] [answered]
I see that stakker is similar but I don't see any place mentioning the difference between stakker and actix, it would also be good to see some benchmarks between them.

**Comment by uazu** [comment, marked as answer]
I have not investigated Actix seriously in order to compare them. Around the time that I was initially developing Stakker, Actix was in the middle of all the unsoundness controversy. The Actix author was rejecting people putting in RefCells to make things safe, and so on. At that time, Actix didn't seem to be a good example to follow. However, I understand that things have improved a lot with Actix since that time.

Bearing that controversy in mind, I made sure the Stakker was sound and efficient from the ground up, so that it would withstand that kind of scrutiny. I use zero-cost statically-checked cell borrows from qcell crate instead of RefCell. (The fundamental approach has recently been proved safe by GhostCell.) I also put work into flat queues that don't need allocations to send and receive messages.  I concentrated on the fundamentals rather than on competing with others. So I expect Stakker to do very well in comparisons with other runtimes. However I don't have a lot of time or motivation to write benchmarks and compare. I'd rather be adding features (async/await integration) or optimising more stuff (e.g. small-Ret optimisation, or timer queue optimisation).

I do agree it might be a selling point to beat Actix on some benchmarks, though. I'd be happy to help out with code/etc if you or anyone else wanted to try that.

**Reply by pickfire** [reply]
Thanks, now I know what is uazu aiming for and it does not seemed much to do with actix although it looks similar.
